src/lib/index.jsx

```diff
@@ -135,6 +141,24 @@ class OtpInput extends Component<Props, State> {
   getOtpValue = () =>
     this.props.value ? this.props.value.toString().split('') : [];
 
+  getPlaceholderValue = () => {
+    const { placeholder, numInputs } = this.props;
+    if (placeholder !== undefined) {
+      if (typeof placeholder !== 'string') {
+        console.error('Placeholder should be a string');
+        return;
+      }
+
+      if (placeholder.length === 1) {
+        return placeholder.repeat(numInputs);
```

Comment:
Since we're returning a placeholder here the subsequent conditions can be replaced with if blocks too instead of else & else if. 

if placeholder.length === 1 return placeholder repeat
if placeholder.length === numInputs return placeholder
warn

A couple of more things caught my attention:
1. Since we have a typeof string check now, the placeholder parent undefined check may be omitted.
2. Let's make the fall-through error message more descriptive. Something like "Length of the placeholder is not equal to the number of inputs."

Reply:
@apollonian 

I have added the requested changes.

1. Refactored the if-else conditions into smaller blocks.
2. Removed placeholder !== undefined condition

3. Error message is changed, but  
> "Length of the placeholder is not equal to the number of inputs." 
is not a valid message, as placeholder can have a length of either 1 or `numInputs`. So I refactored the error message as
> "Invalid placeholder length"

Reply:
Great, thanks!